### PR TITLE
Fixes VSTS 618239: Fatal error editing appsettings.json in new ASP.NE…

### PR DESCRIPTION
**Pull request by mkrueger**
…T Core Web App from template

https://devdiv.visualstudio.com/DevDiv/_workitems/edit/618239

Only the exception from the log but it may be a follow up.

### CODE REVIEW

**Comment by mkrueger** [comment]
@monojenkins backport d15-7

**Comment by KirillOsenkov** [comment]
My recommendation would be to introduce a general purpose extension method such as this:

```
        public static SnapshotPoint? TryGetSnapshotPoint(this ITextSnapshot snapshot, int lineNumber, int columnIndex)
        {
            if (TryGetSnapshotPoint(snapshot, lineNumber, columnIndex, out var snapshotPoint))
            {
                return snapshotPoint;
            }

            return null;
        }

        public static bool TryGetSnapshotPoint(this ITextSnapshot snapshot, int lineNumber, int columnIndex, out SnapshotPoint position)
        {
            position = new SnapshotPoint();

            if (lineNumber < 0 || lineNumber >= snapshot.LineCount)
            {
                return false;
            }

            var line = snapshot.GetLineFromLineNumber(lineNumber);
            if (columnIndex < 0 || columnIndex >= line.Length)
            {
                return false;
            }

            int result = line.Start.Position + columnIndex;
            position = new SnapshotPoint(snapshot, result);
            return true;
        }
```

**Comment by mkrueger** [comment]
Added extension methods - ok now I think I know the real source of the bug in monodevelop line/colums are 1 based.

**Comment by neerajdeshmukh** [comment]
With reference to the scenario shared in the Bug.
I can confirm that the issue is not reproducible on the below builds.
Hence marking this bus as QA Approved.

d15-7 Build and other details:

Build Info: 

=== Visual Studio Enterprise 2017 for Mac ===
Version 7.5.2 (build 19)

=== Xamarin.Mac ===
Version: 4.5.0.324 (Visual Studio Enterprise)
=== Xamarin.iOS ===
Version: 11.11.0.324 (Visual Studio Enterprise)
=== Xamarin.iOS ===
Version: 8.3.99.12 (Visual Studio Enterprise)


Gist Link: https://gist.github.com/parasAmbhore/b35246e298f88725ec6d2ac71960f885

Screen-Cast Link: http://recordit.co/TLaHH4qaXw


Master Build and other details:

Build Info:

=== Visual Studio Enterprise 2017 for Mac (Preview) ===
Version 7.6 Preview (7.6 build 857)

=== Xamarin.Mac ===
Version: 4.5.0.324 (Visual Studio Enterprise)
=== Xamarin.iOS ===
Version: 11.11.0.324 (Visual Studio Enterprise)
=== Xamarin.Android ===
Version: 8.3.99.12 (Visual Studio Enterprise)


Gist Link: https://gist.github.com/parasAmbhore/6dc21d35ee73a07591f28e467fde7339

Screen-Cast Link: http://recordit.co/xOv2SsaycT

Kindly suggest if anything other scenarios need to be tested.

File: main/src/addins/MonoDevelop.SourceEditor2/VSEditor/MDUtils.cs [hunk 1]
```diff
@@ -224,5 +224,45 @@ public static class ClassificationTypeNames
...
+		/// Tries to get a snapshot point out of a monodevelop line/column pair.
+		/// </summary>
+		/// <returns><c>true</c>, if get snapshot point was set, <c>false</c> otherwise.</returns>
+		/// <param name="snapshot">The snapshot.</param>
+		/// <param name="line">Line number (1 based).</param>
+		/// <param name="column">Column number (1 based).</param>
+		/// <param name="snapshotPoint">The snapshot point if return == true.</param>
+		public static bool TryGetSnapshotPoint (this ITextSnapshot snapshot, int line, int column, out SnapshotPoint snapshotPoint)
+		{
+			if (line < 1 || line >= snapshot.LineCount) {
```
Comment (KirillOsenkov) [inline main/src/addins/MonoDevelop.SourceEditor2/VSEditor/MDUtils.cs]: @mkrueger should this be `line > snapshot.LineCount`?

**Pull request merged**